### PR TITLE
Adding fake Wazuh-DB and Analysisd servers to vulnerability detector test tool

### LINKED ISSUE #20762: Create socket mocks for Wazuh-DB and analysisd in the vulnerability detector test tools
## Description

The new vulnerability detector has some inner test tools to verify the components behavior.  But both the Wazuh-DB and analysisd tools require to have a real Wazuh installation to be used.

It's required to mock these sockets to make it possible to run tests without a full Wazuh deployement.

## Subtasks

- [ ] Development. ETA ->
- [ ] Testing. ETA->
- [ ] Final review. ETA->

## DoD

- [ ] Mock the real sockets to allow testing data to be injected in the tests
- [ ] Add UT
- [ ] CI Pass

### PR DESCRIPTION
**Pull request by pereyra-m**
|Related issue|
|---|
|Closes #20762|

## Description

This PR adds new flags to the vulnerability detector test tool to allow the testing without deploying a full Wazuh server
- The fake alerts server will print the messages received
- The fake Wazuh-DB server is able to read a JSON file with fake data and return it to the client

## Tests

The fake report server prints the alerts

The fake Wazuh-DB server receives the queries and the fake OS data is present in the indexed data

- Compilation without warnings in every supported platform
  - [x] Linux

### CODE REVIEW

File: src/wazuh_modules/vulnerability_scanner/testtool/scanner/fakeAgentData/agentData.json
[the remark this answers is not in the document]
Comment (pereyra-m) [reply src/wazuh_modules/vulnerability_scanner/testtool/scanner/fakeAgentData/agentData.json:1]: Hi @GabrielEValenzuela !

The `TC-xxx` folders are meant to contain fake events from the agents and the purpose of this folder is completely different: it acts like an agent DB stored in Wazuh-DB to retrieve OS data.

We might change its name if the purpose isn't clear, but I wouldn't use the same as the other.

File: src/wazuh_modules/vulnerability_scanner/testtool/scanner/main.cpp
Comment (GabrielEValenzuela) [inline src/wazuh_modules/vulnerability_scanner/testtool/scanner/main.cpp:1]: Could you improve the help information?

File: src/wazuh_modules/vulnerability_scanner/testtool/scanner/fakeAgentData/agentData.json
[the remark this answers is not in the document]
Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/testtool/scanner/fakeAgentData/agentData.json:1]: Don't worry! I see now how is the logic and I agree with you! Thanks @pereyra-m !

File: src/wazuh_modules/vulnerability_scanner/testtool/scanner/main.cpp
Comment (pereyra-m) [reply src/wazuh_modules/vulnerability_scanner/testtool/scanner/main.cpp:1]: @GabrielEValenzuela Done!

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**